### PR TITLE
LibJS/Bytecode: Ensure iterator is closed after array destructuring

### PR DESCRIPTION
**Pull request by awesomekling**
If destructuring didn't exhaust the iterator, we have to manually close the iterator.

155 new passes on test262. :^)

### CODE REVIEW

File: Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp [hunk 1]
```diff
@@ -1306,6 +1308,18 @@ static Bytecode::CodeGenerationErrorOr<void> generate_array_binding_pattern_byte
...
+
+    generator.emit<Bytecode::Op::Load>(is_iterator_exhausted_register);
+    generator.emit<Bytecode::Op::JumpConditional>().set_targets(
+        Bytecode::Label { done_block },
+        Bytecode::Label { not_done_block });
+
+    generator.switch_to_basic_block(not_done_block);
+    generator.emit<Bytecode::Op::Load>(iterator_reg);
+    generator.emit<Bytecode::Op::IteratorClose>(Completion::Type::Normal, Optional<Value> {});
+
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:1322]: do we need to switch to the `done_block` here?

Comment (Hendiadyoin1) [reply Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:1322]: /jump back to it

File: Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp [hunk 2]
```diff
@@ -1306,6 +1308,19 @@ static Bytecode::CodeGenerationErrorOr<void> generate_array_binding_pattern_byte
...
+    generator.switch_to_basic_block(done_block);
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:1323]: Now it still seems execution ends[1] when entering the `not_done` block

[1]: By reaching the end of a block without doing any control flow through the terminator

Comment (awesomekling) [reply Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp:1323]: Yeah that's clearly wrong. 😅 
And if we fix that, a bunch of tests break... back to the drawing board!

**Pull request merged**